packages/@lwc/template-compiler/src/codegen/expression.ts

```diff
@@ -215,17 +216,23 @@ export function bindAttributeExpression(
         return codeGen.genScopedFragId(expression);
     }
     if (isSvgUseHref(elmName, attrName, namespace)) {
+        // Apply the fragment id scoping transformation if necessary.
+        // This scoping can be skipped if the value is a string literal that doesn't start with a "#"
+        const value =
+            isStringLiteral(attrValue) && !isFragmentOnlyUrl(attrValue.value)
+                ? t.literal(attrValue.value)
+                : codeGen.genScopedFragId(expression);
```

Comment:
This is a little kooky to me – basically we only add the `api_scoped_frag_id()` function when the value is a non-literal or starts with a `#`:

```js
// dynamic
api_scoped_frag_id($cmp.foo)

// static, starts with #
api_scoped_frag_id('#yolo')

// static, does not start with #
'/foo.svg'
```

We _could_ just run everything through the `api_scoped_frag_id` function, since it already checks to see if the value starts with a `#`. But I guess this is an optimization for static values that don't start with `#`. So I preserved the existing behavior.